### ISSUE TITLE
Fail when recoverable parser errors occurs in ignored files with rustfmt-recursive 

### ISSUE THREAD

**Issue by sasurau4** [rust-lang/rustfmt#4592]
**Describe the bug**

Fail when formatting recoverable parser errors occurs in ignored files with rustfmt-recursive.
I found it working #4113 with `issue-3779/ice.rs` 

**To Reproduce**

1. Add `// rustfmt-recursive: true` to the head of `tests/source/issue-3779/lib.rs`
2. run `cargo make test`
3. Fail with following error

```
error: invalid suffix `x` for integer literal
 --> tests/source/issue-3779/ice.rs:2:5
  |
2 |     1x;
  |     ^^ invalid suffix `x`
  |
  = help: the suffix must be one of the integral types (`u32`, `isize`, etc)
```

**Expected behavior**

`cargo make test` passes

**Meta**

- rustfmt version: current master 81ad114d12c2d7dd2253afd64f0d2019697aa6de

**Comment by calebcartwright** [comment]
Haven't had time to look at this closely but based on a quick glance my sense is that this is actually failing because the config/ignore settings aren't actually getting picked up, so rustfmt is attempting to parse the ice mod?

**Comment by sasurau4** [comment]
@calebcartwright Yes, you are right. rustfmt is attempting to parse the ice mod. It will be fixed by #4596
